dds/DCPS/transport/rtps_udp/RtpsUdpSendStrategy.cpp

```diff
@@ -268,8 +268,9 @@ RtpsUdpSendStrategy::send_multi_i(const iovec iov[], int n,
 {
   ssize_t result = -1;
   typedef AddrSet::const_iterator iter_t;
+  const NetworkAddress empty = NetworkAddress();
   for (iter_t iter = addrs.begin(); iter != addrs.end(); ++iter) {
-    if (*iter == NetworkAddress()) {
+    if (*iter == empty) {
```

Comment:
Grepping it looks like we do checks like this in several places, would it make sense to add a boolean overload to `NetworkAddress` instead of comparing it to an empty object that has to be constructed each time?

Reply:
That's what this was intended to do, though I suppose you're asking about caching the empty value in a more globally accessible location. I'll add `operator bool()` and `bool operator!()` for `NetworkAddress` but I don't think this is right PR to switch everything over to using them.